themes/default/subtemplates/posting.inc.tpl

```diff
@@ -82,61 +82,61 @@
 {if $form_type==0}
 <fieldset>
 
-<p>
+<div>
 <label for="name" class="input">{#name_marking#}</label>
 <input id="name" type="text" size="40" name="{$fld_user_name}" value="{if $name}{$name}{/if}" maxlength="{$settings.username_maxlength}"  tabindex="1" />
-</p>
+</div>
 
-<p>
+<div>
 <label for="email" class="input">{#email_marking#}</label>
-<input id="email" type="text" size="40" name="{$fld_user_email}" value="{if $email}{$email}{/if}" maxlength="{$settings.email_maxlength}" tabindex="2" />&nbsp;<span class="xsmall">{#optional_email#}</span>
-</p>
+<input id="email" type="email" size="40" name="{$fld_user_email}" value="{if $email}{$email}{/if}" maxlength="{$settings.email_maxlength}" tabindex="2" />&nbsp;<span class="xsmall">{#optional_email#}</span>
+</div>
 
-<p class="hp">
+<div class="hp">
 <label for="repeat_email" class="main">{#honeypot_field_marking#}</label>
-<input id="repeat_email" type="text" size="40" name="{$fld_repeat_email}" value="{if $honey_pot_email}{$honey_pot_email}{/if}" maxlength="{$settings.email_maxlength}" tabindex="-1" />
-</p>
+<input id="repeat_email" type="email" size="40" name="{$fld_repeat_email}" value="{if $honey_pot_email}{$honey_pot_email}{/if}" maxlength="{$settings.email_maxlength}" tabindex="-1" />
+</div>
 
-<p>
+<div>
 <label for="hp" class="input">{#hp_marking#}</label>
-<input id="hp" type="text" size="40" name="{$fld_hp}" value="{if $hp}{$hp}{/if}" maxlength="{$settings.hp_maxlength}" tabindex="3" />&nbsp;<span class="xsmall">{#optional#}</span>
-</p>
+<input id="hp" type="url" size="40" name="{$fld_hp}" value="{if $hp}{$hp}{/if}" maxlength="{$settings.hp_maxlength}" tabindex="3" />&nbsp;<span class="xsmall">{#optional#}</span>
```

Comment:
inside the label?

Reply:
Oha!

Reply:
Let's take a look to the "tags" input field, the code is (currently)

```
<div>
<label for="tags" class="input">{#tags_marking#} <span class="xsmall">{#tags_note#}</span></label>
<input id="tags" type="text" size="50" name="tags" value="{$tags|default:""}" maxlength="253" tabindex="-1" />
</div>
```

The first _xsmall span_ is inside the label tag. If we compare this code to the e.g. location input, i.e.,

```
<div>
<label for="location" class="input">{#location_marking#}</label>
<input id="location" type="text" size="40" name="{$fld_location}" value="{if $location}{$location}{/if}" maxlength="{$settings.location_maxlength}" tabindex="4" />&nbsp;<span class="xsmall">{#optional#}</span>
</div>
```

you will notice that the _xsmall span_ is not inside the label tag, but behind the input.   I wonder why the two pieces of code are different.

Reply:
Ok, now I understand your concern. I will correct the code.